### PR TITLE
Windows build/test using CircleCI

### PR DESCRIPTION
**Pull request by rcoringrato-sfdc**
### What does this PR do?

- Add build/test job for windows
- Process and upload code coverage

### What issues does this PR fix or reference?

@W-7129488@

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -47,6 +46,41 @@ jobs:
...
 
+  run-win-tests:
+    description: 'Run tests on Windows'
+    executor: win/default
+    steps:
+      - run: git config --global core.autocrlf false
+      - checkout
+      - restore_cache:
+          keys:
+            - dependency-cache-win1-{{checksum ".circleci/config.yml"}}-{{checksum "yarn.lock"}}
```
Comment (lcampos) [inline .circleci/config.yml]: What's the benefit of including the one for the circleci config ? Isn't the checksum for yarn.lock enough ?

File: .circleci/config.yml [hunk 2]
```diff
@@ -47,6 +46,41 @@ jobs:
...
+            # if cache for exact version of 'package.json' is not present, then load the most recent cache
+            - dependency-cache-win1-
+      - run:
+          name: 'Install dependencies'
+          command: |
+            yarn --version
+            node --version
+            yarn
+            npm install -g codecov
```
Comment (lcampos) [inline .circleci/config.yml]: Can we change this to `yarn global ...` or is there a reason why it needs to be installed using npm ?

File: package.json [hunk 3]
```diff
@@ -69,7 +69,7 @@
   },
   "repository": "forcedotcom/salesforcedx-templates",
   "scripts": {
-    "clean:lib": "node scripts/remove-lib",
+    "clean:lib": "node scripts/remove-lib && shx rm -rf coverage && shx rm -rf .nyc_output",
```
Comment (lcampos) [inline package.json]: Not a thing we need to address in this PR but should we get rid of the `remove-lib` script and replace it with `shx` ? If so I can create a work item for it.

File: .circleci/config.yml (hunk 1 above)
Comment (rcoringrato-sfdc) [reply .circleci/config.yml]: @lcampos A benefit is that you start with a clean cache when you make circleci config changes.

File: .circleci/config.yml (hunk 2 above)
Comment (rcoringrato-sfdc) [reply .circleci/config.yml]: @lcampos No reason other than habit. I'll change it.

File: package.json (hunk 3 above)
Comment (rcoringrato-sfdc) [reply package.json]: @lcampos Straight forward change, I'll include in this PR.

File: .circleci/config.yml (hunk 1 above)
Comment (lcampos) [reply .circleci/config.yml]: Got it 👍

**Pull request merged**